nuxeo-core/nuxeo-core-storage-dbs/src/main/java/org/nuxeo/ecm/core/storage/dbs/DBSSession.java

```diff
@@ -1454,8 +1454,8 @@ protected static ACP memToAcp(Serializable acpSer) {
             String idKey = idKeyHolder.getValue();
             List<String> ids = new ArrayList<>(pl.size());
             for (Map<String, Serializable> map : pl) {
-                String id = (String) map.get(idKey);
-                ids.add(id);
+                Serializable id = map.get(idKey);
+                if (id != null) ids.add(String.valueOf(id));
```

Comment:
We always use `{}` blocks even for very short conditions:

    if (id != null) {
        ids.add(String.valueOf(id));
    }

However I'd keep the cast `(String) id` instead of `String.valueOf` as the key should already have been turned into a `String` by lower layers, and I don't want to hide an unexpected change in behavior by doing `String.valueOf`

Reply:
I add this because a Long (major_version) was failing to cast to a String.  That was halfway through my fix so could have been a result of something else.  Let me take this out and see if it still breaks.